app/src/common/shared/org/mozilla/vrbrowser/SessionStore.java

```diff
@@ -72,10 +68,151 @@ public static SessionStore get() {
         GeckoSession mSession;
     }
 
+    private static class GeckoError {
+
+        public static final String UNABLE_TO_CONNECT_ERROR_URL = "resource://android/assets/html/error/unable-to-connect.html";
+        public static final String FILE_NOT_FOUND_ERROR_URL = "resource://android/assets/html/error/file-not-found.html";
+        public static final String UNTRUSTED_CONNECTION_ERROR_URL = "resource://android/assets/html/error/this-connection-is-untrusted.html";
+        public static final String SECURE_CONNECTION_ERROR_URL = "resource://android/assets/html/error/secure-connection-failed.html";
+
+        public static final String NET_ERROR_TYPE = "about:neterror";
+        public static final String CERT_ERROR_TYPE = "about:certerror";
+
+        public enum GeckoErrorType {
+            UNKNOWN_ERROR,
+            NET_ERROR,
+            CERT_ERROR
+        }
+
+        public static final String DNSNOTFOUND_ERROR_SUBTYPE = "dnsNotFound";
+        public static final String NSSFAILURE2_ERROR_SUBTYPE = "nssFailure2";
+        public static final String FILENOTFOUND_ERROR_SUBTYPE = "fileNotFound";
+        public static final String NSSBADCERT_ERROR_SUBTYPE = "nssBadCert";
+
```

Comment:
these are four of the many others I found in Gecko's [`netError.dtd`](https://dxr.mozilla.org/mozilla-central/source/dom/locales/en-US/chrome/netError.dtd):

- `blockedByPolicy`
- `connectionFailure`
- `contentEncodingError`
- `corruptedContentErrorv2`
- `cspBlocked`
- `deniedPortAccess`
- `dnsNotFound`
- `fileAccessDenied`
- `fileNotFound`
- `generic`
- `inadequateSecurityError`
- `malformedURI`
- `netInterrupt`
- `netOffline`
- `netReset`
- `netTimeout`
- `notCached`
- `nssBadCert`
- `nssFailure2`
- `proxyConnectFailure`
- `proxyResolveFailure`
- `redirectLoop`
- `remoteXUL`
- `unknownProtocolFound`
- `unknownSocketType`
- `unsafeContentType`

I assume those should be handled as well (or a follow-up issue filed to handle the subtypes dynamically like they are in Gecko's [`netError.xhtml`](https://dxr.mozilla.org/mozilla-central/source/docshell/resources/content/netError.xhtml))

Reply:
Updated to handle all the subtypes